include/gul17/join_split.h

```diff
@@ -177,7 +177,7 @@ split(std::string_view text, const std::regex& delimiter,
     auto const end = std::cregex_iterator{ };
     auto result = StringContainer{ };
 
-    auto parts = std::cregex_iterator(text.begin(), text.end(), delimiter);
+    auto parts = std::cregex_iterator(text.data(), text.data() + text.size(), delimiter);
```

Comment:
```suggestion
    auto parts = std::regex_iterator<std::string_view::iterator>(text.data(), text.data() + text.size(), delimiter);
```

Normally the regexes should work with string_view, and if you use the string-view regex-iterator this line at least compiles. I'm not sure on how much the code further down depends on c-string regmatches etc, but they can/should probably just all be switched to `svregex_iterator` etc pp.

The cast to string_view further down should be not needed then anymore, probably.

I could in principle also check that but I have not really a Windows machine handy, and testing at Godbold is tedious.